pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -195,6 +199,33 @@ private Calendar getCalendar(int sign, int hr, int min, int sec) {
     NEGATIVE;
   }
 
+  /**
+   * Implements the rounding logic of the driver.
+   *
+   * <p>Truncates to microsecond (Postgres resolution) and rounds up
+   * if the nanosecond part is above 499.</p>
+   */
+  static final class RoundToMicros implements TemporalAdjuster {
+
+    @Override
+    public Temporal adjustInto(Temporal temporal) {
+      int nanosOfSecond = temporal.get(ChronoField.NANO_OF_SECOND);
+      int microsOfSecond = temporal.get(ChronoField.MICRO_OF_SECOND);
+      int remainderNanos = nanosOfSecond - microsOfSecond * 1000;
```

Comment:
Would it be better to use `int remainderNanos = nanosOfSecond % 1000;`?

Reply:
I believe yes because it would have shortened the code and allowed us to get rid of the `microsOfSecond` variable. After thinking about it for some time more I decided to factor out the reminder nanos logic into its own `TemporalField`. I believe it helps readability even more but let me know if you feel that it is overkill and we should keep with

    int remainderNanos = nanosOfSecond % 1000;